keycloak/lib/stacks/utils.ts

```diff
@@ -0,0 +1,120 @@
+/**
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+import { Stack, StackProps } from 'aws-cdk-lib';
+import {
+  CertificateAuthority, CfnCertificate, CfnCertificateAuthority, CfnCertificateAuthorityActivation,
+} from 'aws-cdk-lib/aws-acmpca';
+import {
+  Certificate, CertificateValidation, KeyAlgorithm, PrivateCertificate,
+} from 'aws-cdk-lib/aws-certificatemanager';
+import { HostedZone } from 'aws-cdk-lib/aws-route53';
+import { Secret } from 'aws-cdk-lib/aws-secretsmanager';
+import { Construct } from 'constructs';
+
+export interface KeycloakUtilsProps extends StackProps {
+  readonly hostedZone: string,
+}
+
+export class KeycloakUtils extends Stack {
+  readonly zone: HostedZone;
+
+  readonly certificateArn: string;
+
+  readonly keycloakDBpassword: Secret;
```

Comment:
nit: `keycloakDbPassword`?

Reply:
Also, should this be part of Db stack? A small password change will result in deployment of 2 stacks.

Reply:
Keeping all the secrets in one place for now. All these secrets are used for multiple stacks. Eg: database and keycloak installation both.